src/optimization/utils/constraints.jl

```diff
@@ -208,8 +208,8 @@ function add_complementary_set_constraint!(m::Model, H::HalfSpace, z::Vector{Var
     @constraint(m, a * z .>= b)
     return nothing
 end
-function add_complementary_set_constraint!(m::Model, PC::PolytopeComplement, z::Vector{VariableRef})
-    add_set_constraint!(m, PC.P, z)
+function add_complementary_set_constraint!(m::Model, PC::Complement{<:Real, <:AbstractPolytope}, z::Vector{VariableRef})
```

Comment:
Can we go with just `Complement`? There shouldn't be a need for the type parameters.

Reply:
Also, since the functions that use `Complement` just unwrap it and pass to the complementary function (which are all strictly typed (HalfSpace, HPolytope, etc.), we will just run into the same method error one layer deeper, which is fine. So we should be able to stick to `::Complement` without a problem.

Reply:
oh got it, I'll add that then.